packages/docusaurus/src/server/plugins/init.ts

```diff
@@ -184,36 +234,41 @@ export default function initPlugins({
     }
   }
 
-  const plugins: InitializedPlugin[] = pluginConfigs
-    .map((pluginConfig) => {
-      if (!pluginConfig) {
-        return null;
-      }
-      const normalizedPluginConfig = normalizePluginConfig(
-        pluginConfig,
-        pluginRequire,
-      );
-      const pluginVersion: DocusaurusPluginVersionInformation =
-        doGetPluginVersion(normalizedPluginConfig);
-      const pluginOptions = doValidatePluginOptions(normalizedPluginConfig);
-
-      // Side-effect: merge the normalized theme config in the original one
-      context.siteConfig.themeConfig = {
-        ...context.siteConfig.themeConfig,
-        ...doValidateThemeConfig(normalizedPluginConfig),
-      };
+  const pluginTypes = ['theme', 'plugin'] as PluginType[];
```

Comment:
```suggestion
  const pluginTypes = ['theme', 'plugin'] as const
```

Reply:
I don't think it would work, because the usage expects the specific union type.

As it won't be exposed externally, we could even convert it to an enum (I haven't done it because our API always exposes union types instead of enums).

Reply:
No, please don't make enums, I can make a whole case against them😅 I don't know why `['theme', 'plugin'] as const` won't work though

Reply:
Ok, I'll cast it to `const` then.